### PR TITLE
add changes for "Define applications but provide flexibility to build applications"

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x] This PR is based on the same version of rt_timing branch. 

- [ ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. 

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

Provide a detailed description of what this PR does. What bug does it fix, or what feature does it add? Is a change of answers expected from this PR? Are any library updates included in this PR (modulefiles etc.)?

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes  [ #416](https://github.com/ufs-community/ufs-weather-model/issues/416)

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first? NO

### CODE REVIEW

File: tests/rt.conf [hunk 1]
```diff
@@ -23,21 +23,21 @@ RUN     | fv3_iau
 RUN     | fv3_lheatstrg                                                                                                           |                                         | fv3 |
 
 # WW3 not working on Cheyenne in the past, need to check if it works now
-COMPILE | SUITES=FV3_GFS_2017,FV3_GFS_2017_gfdlmp WW3=Y                                                                           | + wcoss_dell_p3 hera.intel orion.intel  | fv3 |
+COMPILE | APP=ATMW SUITES=FV3_GFS_2017,FV3_GFS_2017_gfdlmp
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: Add `| + wcoss_dell_p3 hera.intel orion.intel  | fv3 |`

File: tests/rt.conf [hunk 2]
```diff
@@ -23,21 +23,21 @@ RUN     | fv3_iau
...
 RUN     | fv3_gfdlmprad                                                                                                           | + wcoss_dell_p3 hera.intel orion.intel  | fv3 |
 RUN     | fv3_gfdlmprad_atmwav                                                                                                    | + wcoss_dell_p3 hera.intel orion.intel  | fv3 |
 RUN     | fv3_wrtGauss_nemsio_c768                                                                                                | +               hera.intel orion.intel  | fv3 |
 
 # Run multigases test in REPRO mode to avoid numerical instability in the deep atmosphere
-COMPILE | SUITES=FV3_GFS_2017_fv3wam 32BIT=Y MULTI_GASES=Y REPRO=Y                                                                |                                         | fv3 |
+COMPILE | APP=ATM SUITES=FV3_GFS_2017_fv3wam 32BIT=Y MULTI_GASES=Y REPRO=Y
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: Add ` |                                         | fv3 |`

File: tests/rt.conf [hunk 3]
```diff
@@ -165,21 +165,21 @@ RUN     | cpld_restart_bmarkfrac
...
 RUN     | cpld_debug                                                                                                              | - wcoss_cray jet.intel                  | fv3 |
 RUN     | cpld_debugfrac                                                                                                          | - wcoss_cray jet.intel                  | fv3 |
 
 ###################################################################################################################################################################################
 # Data Atmosphere tests                                                                                                                                                           #
 ###################################################################################################################################################################################
 
-COMPILE | DATM=Y S2S=Y                                                                                                            | - wcoss_cray jet.intel                  | fv3 |
+COMPILE | APP=DATM_NEMS
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: Add` | - wcoss_cray jet.intel                  | fv3 |`

File: tests/rt.conf [hunk 4]
```diff
@@ -190,5 +190,5 @@ RUN     | datm_bulk_gefs
 RUN     | datm_mx025_cfsr                                                                                                         | - wcoss_cray jet.intel gaea.intel       | fv3 |
 RUN     | datm_mx025_gefs                                                                                                         | - wcoss_cray jet.intel                  | fv3 |
 
-COMPILE | DATM=Y S2S=Y DEBUG=Y                                                                                                    | - wcoss_cray jet.intel                  | fv3 |
+COMPILE | APP=DATM_NEMS
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: Add | - wcoss_cray jet.intel                  | fv3 |

File: CMakeLists.txt [hunk 5]
```diff
@@ -342,18 +395,21 @@ target_link_libraries(ufs_model PRIVATE ufs
...
 ### Install
 ###############################################################################
-install(
-  TARGETS fms
-  EXPORT fms-config
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
-  PUBLIC_HEADER DESTINATION include )
+if(FMS)
+  install(
```
Comment (DusanJovic-NOAA) [inline CMakeLists.txt]: This block will be removed in https://github.com/ufs-community/ufs-weather-model/pull/372

File: CMakeLists.txt [hunk 6]
```diff
@@ -125,82 +155,71 @@ target_link_libraries(nemsio::nemsio INTERFACE w3emc::w3emc_d bacio::bacio_4)
...
-
-list(APPEND _fms_defs_public use_libMPI
-                             use_netCDF
-                             GFS_PHYS
-                             INTERNAL_FILE_NML)
-if(QUAD_PRECISION)
-  list(APPEND _fms_defs_public ENABLE_QUAD_PRECISION)
-endif()
+if(FMS)
+  include(fms_files.cmake)
```
Comment (DusanJovic-NOAA) [inline CMakeLists.txt]: This block will be removed in ufs-community#372

File: tests/rt.conf (hunk 1 above)
Comment (aerorahul) [reply tests/rt.conf]: addressed in 277cefb

File: tests/rt.conf (hunk 2 above)
Comment (aerorahul) [reply tests/rt.conf]: addressed in 277cefb

File: tests/rt.conf (hunk 3 above)
Comment (aerorahul) [reply tests/rt.conf]: addressed in 277cefb

File: tests/rt.conf (hunk 4 above)
Comment (aerorahul) [reply tests/rt.conf]: addressed in 277cefb

File: CMakeLists.txt (hunk 6 above)
Comment (aerorahul) [reply CMakeLists.txt]: It is simplified in #372 but not removed.

File: CMakeLists.txt (hunk 5 above)
Comment (aerorahul) [reply CMakeLists.txt]: addressed in 3624511

**Pull request merged**